AndroidSDKTests/src/test/java/com/leanplum/LeanplumInboxTest.java

```diff
@@ -48,141 +49,191 @@
+      public void onForceContentUpdate(boolean success) {
+        assertFalse(success);
+      }
+    };
+
+    // Add synced callback  to be able to validate.
+    Leanplum.getInbox().addSyncedHandler(inboxSyncedCallbackFalse);
+
+    // Return false when checks for internet connection.
+    doReturn(false).when(Util.class, "isConnected");
+
+    // Download messages.
+    Leanplum.getInbox().downloadMessages();
+
+    //  Remove synced callback afterwards so we don't get synced callbacks anymore.
+    Leanplum.getInbox().removeSyncedHandler(inboxSyncedCallbackFalse);
+  }
+
+  @Test
+  public void testDisablePrefetching() {
+    leanplumInbox.disableImagePrefetching();
+    assertFalse(leanplumInbox.isInboxImagePrefetchingEnabled());
+  }
+
+  @Test
+  public void testInvalidMessageIdIsRejected() {
+    Date delivery = new Date(100);
+    Date expiration = new Date(200);
+    HashMap<String, Object> map = new HashMap<>();
+    map.put(Constants.Keys.MESSAGE_DATA, new HashMap<String, Object>());
+    map.put(Constants.Keys.DELIVERY_TIMESTAMP, delivery.getTime());
+    map.put(Constants.Keys.EXPIRATION_TIMESTAMP, expiration.getTime());
+    map.put(Constants.Keys.IS_READ, true);
+
+    LeanplumInboxMessage invalidMessage = LeanplumInboxMessage.createFromJsonMap("messageId", map);
+    assertNull(invalidMessage);
+  }
+
+  @Test
+  public void testRemoveMessageFromInbox() {
+    Date delivery = new Date(100);
+    Date expiration = new Date(200);
+    HashMap<String, Object> map = new HashMap<>();
+    map.put(Constants.Keys.MESSAGE_DATA, new HashMap<String, Object>());
+    map.put(Constants.Keys.DELIVERY_TIMESTAMP, delivery.getTime());
+    map.put(Constants.Keys.EXPIRATION_TIMESTAMP, expiration.getTime());
+    map.put(Constants.Keys.IS_READ, true);
+    ResponseHelper.seedResponse("/responses/newsfeed_response.json");
+
+    leanplumInbox.downloadMessages();
+    leanplumInbox.removeMessage("5231495977893888##1");
+
+    assertEquals(1, leanplumInbox.count());
+  }
+
+  @Test
+  public void testInboxRemovesACachedMessageAfterDownloading() {
+    Date delivery = new Date(100);
+    Date expiration = new Date(200);
+    HashMap<String, Object> map = new HashMap<>();
+    map.put(Constants.Keys.MESSAGE_DATA, new HashMap<String, Object>());
+    map.put(Constants.Keys.DELIVERY_TIMESTAMP, delivery.getTime());
+    map.put(Constants.Keys.EXPIRATION_TIMESTAMP, expiration.getTime());
+    map.put(Constants.Keys.IS_READ, true);
+    LeanplumInboxMessage message1 = LeanplumInboxMessage.createFromJsonMap("messageId##1", map);
+    Map<String, LeanplumInboxMessage> messages = new HashMap<>();
+    messages.put("message##1", message1);
+    leanplumInbox.update(messages, 1, true);
+    ResponseHelper.seedResponse("/responses/newsfeed_response.json");
+
+    assertEquals(1, leanplumInbox.count());
+
+    leanplumInbox.downloadMessages();
+
+    assertEquals(2, leanplumInbox.count());
+  }
+
+  @Test
+  public void testGettingUnreadMessagesAfterARead() {
```

Comment:
is this test implemented here  https://github.com/Leanplum/Leanplum-Android-SDK/pull/164/files#diff-d4fac6525f945e4a092843223e65630dR132?